beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -196,41 +198,82 @@ func (s *Service) hasPeer() bool {
 var errNoPeersForPending = errors.New("no suitable peers to process pending block queue, delaying")
 
 // processAndBroadcastBlock validates, processes, and broadcasts a block.
-// part of the function is to request missing blobs from peers if the block contains kzg commitments.
-func (s *Service) processAndBroadcastBlock(ctx context.Context, b interfaces.ReadOnlySignedBeaconBlock, blkRoot [32]byte) error {
+// Part of the function is to request missing sidecars from peers if the block contains kzg commitments.
+func (s *Service) processAndBroadcastBlock(ctx context.Context, b interfaces.ReadOnlySignedBeaconBlock, blkRoot [fieldparams.RootLength]byte) error {
+	blockSlot := b.Block().Slot()
+
 	if err := s.validateBeaconBlock(ctx, b, blkRoot); err != nil {
 		if !errors.Is(ErrOptimisticParent, err) {
-			log.WithError(err).WithField("slot", b.Block().Slot()).Debug("Could not validate block")
+			log.WithError(err).WithField("slot", blockSlot).Debug("Could not validate block")
 			return err
 		}
 	}
 
-	request, err := s.pendingBlobsRequestForBlock(blkRoot, b)
+	blockEpoch, denebForkEpoch, fuluForkEpoch := slots.ToEpoch(blockSlot), params.BeaconConfig().DenebForkEpoch, params.BeaconConfig().FuluForkEpoch
+
+	roBlock, err := blocks.NewROBlockWithRoot(b, blkRoot)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "new ro block with root")
 	}
-	if len(request) > 0 {
-		peers := s.getBestPeers()
-		peerCount := len(peers)
-		if peerCount == 0 {
-			return errors.Wrapf(errNoPeersForPending, "block root=%#x", blkRoot)
+
+	if blockEpoch >= fuluForkEpoch {
+		if err := s.requestAndSaveMissingDataColumnSidecars([]blocks.ROBlock{roBlock}); err != nil {
+			return errors.Wrap(err, "request and save missing data column sidecars")
+		}
+
+		if err := s.receiveAndBroadCastBlock(ctx, b, blkRoot, blockSlot); err != nil {
```

Comment:
we have this copied 3 times. maybe we should do if + else if and just have the receive and broadcast block at the bottom